modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/field-sets/FieldSetModal.es.js

```diff
@@ -46,18 +48,71 @@ const ModalContent = ({
 		state: {},
 	});
 	const [dataLayoutIsEmpty, setDataLayoutIsEmpty] = useState(true);
+	const [editingLanguageId, setEditingLanguageId] = useState(
+		defaultLanguageId
+	);
 	const [name, setName] = useState({});
+
+	const {contentType} = appProps;
 	const {
 		dataLayoutBuilder,
 		dispatch,
-		state: {dataLayout},
+		state: {dataDefinition, dataLayout},
 	} = childrenContext;
 
-	const {contentType} = appProps;
+	const actionProps = {
+		availableLanguageIds: dataDefinition?.availableLanguageIds,
+		childrenContext,
+		defaultLanguageId,
+		fieldSet,
+	};
+
+	const createFieldSet = useCreateFieldSet(actionProps);
+	const saveFieldSet = useSaveFieldSet(actionProps);
+	const propagateFieldSet = usePropagateFieldSet();
+
+	const onSave = () => {
+		const hasRemovedField = () => {
+			const fieldNames = fieldSet.dataDefinitionFields.map(
+				({name}) => name
+			);
 
-	const availableLanguageIds = [
-		...new Set([...Object.keys(name), defaultLanguageId]),
-	];
+			const [prevLayoutFields, actualLayoutFields] = [
```

Comment:
I know, we use it before, but could we rename `actualLayoutFields ` to `currentLayoutFields`?

Reply:
Done